src/core/usecases/initScriptsGenerator.ts

```diff
@@ -0,0 +1,73 @@
+import type { ThunkAction } from "../setup";
+import { createSlice } from "@reduxjs/toolkit";
+import { id } from "tsafe/id";
+import { K8sClusterParams } from "./launcher";
+import { useMemo } from "react";
+
+export const exportK8sCredentials = [
+    {
+        id: "bash",
+        label: "bash",
+        fileName: "init-kub.sh",
+        text: (c: K8sClusterParams) =>
+            `#!/bin/bash
+kubectl config set-cluster ${c.K8S_CLUSTER} --server=${c.K8S_SERVER_URL}
+kubectl config set-credentials ${c.K8S_USER} --token ${c.K8S_TOKEN} 
+kubectl config set-context ${c.K8S_CLUSTER} --user=${c.K8S_USER} --cluster=${c.K8S_CLUSTER} --namespace=${c.K8S_NAMESPACE}
+kubectl config use-context ${c.K8S_CLUSTER}
+`,
+    },
+    {
+        id: "shell",
+        label: "shell",
+        fileName: "init-kub.sh",
+        text: (c: K8sClusterParams) =>
+            `#!/bin/sh
+kubectl config set-cluster ${c.K8S_CLUSTER} --server=${c.K8S_SERVER_URL}
+kubectl config set-credentials ${c.K8S_USER} --token ${c.K8S_TOKEN} 
+kubectl config set-context ${c.K8S_CLUSTER} --user=${c.K8S_USER} --cluster=${c.K8S_CLUSTER} --namespace=${c.K8S_NAMESPACE}
+kubectl config use-context ${c.K8S_CLUSTER}
+`,
+    },
+];
+
+type InitScriptGeneratorState = string | null;
```

Comment:
The state shoud be more descriptive.  
We should use this slice for generating the other slices, not only the Kubernetes one.